batch7crud-roll4/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDao.java

```diff
@@ -0,0 +1,126 @@
+                e.setStation(rs.getString("station"));
+                employeeList.add(e);
+            }
+            return employeeList;
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+    }
+
+    public Employee fetchById(int id) throws DataException {
+        try {
+            Employee e = null;
+            String sql = "SELECT * FROM employees WHERE id=?";
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            stmt.setInt(1, id);
+            ResultSet rs = stmt.executeQuery();
+
+            while (rs.next()) {
+                e = new Employee();
+                e.setId(rs.getInt("id"));
+                e.setFirstName(rs.getString("first_name"));
+                e.setLastName(rs.getString("last_name"));
+                e.setStation(rs.getString("station"));
+            }
+            return e;
+
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+    }
+
+    public void insert(Employee employee) throws DataException {
+        try {
+            String sql = "INSERT INTO employees(first_name, last_name, station) VALUES(?,?,?)";
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            stmt.setString(1, employee.getFirstName());
+            stmt.setString(2, employee.getLastName());
+            stmt.setString(3, employee.getStation());
+
+            stmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+    }
+
+    public void update(Employee employee) throws DataException {
+        try {
+            String sql = "UPDATE employees SET first_name=?, last_name=?, station=? WHERE id=?";
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            stmt.setString(1, employee.getFirstName());
+            stmt.setString(2, employee.getLastName());
+            stmt.setString(3, employee.getStation());
+            stmt.setInt(4, employee.getId());
+
+            stmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+    }
+
+    public void deleteEmployee(int id) throws DataException {
+        try {
+            String sql = "DELETE FROM employees WHERE id=?";
+
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            stmt.setInt(1, id);
+            stmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+    }
+
+    public int fetchNoOfRecords() throws DataException {
+        try {
+            int noOfRecords = 0;
+            String sql = "SELECT COUNT(*) FROM employees";
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            ResultSet rs = stmt.executeQuery();
+            while (rs.next()) {
+                noOfRecords = rs.getInt(1);
+            }
+            return noOfRecords;
+
+        } catch (SQLException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)